akeneo-design-system/src/components/Table/TableHeaderCell/TableHeaderCell.unit.tsx

```
@@ -0,0 +1,137 @@
+      </Table.Header>
+    </Table>
+  );
+
+  const header = screen.getByText('An header');
+  fireEvent.click(header);
+
+  expect(onDirectionChange).toBeCalledWith('ascending');
+});
+
+test('it calls onDirectionChange handler when user clicks on sortable descending cell', () => {
+  const onDirectionChange = jest.fn();
+  render(
+    <Table>
+      <Table.Header>
+        <Table.HeaderCell isSortable={true} onDirectionChange={onDirectionChange} sortDirection="descending">
+          An header
+        </Table.HeaderCell>
+        <Table.HeaderCell>Another header</Table.HeaderCell>
+      </Table.Header>
+    </Table>
+  );
+
+  const header = screen.getByText('An header');
+  fireEvent.click(header);
+
+  expect(onDirectionChange).toBeCalledWith('ascending');
+});
+
+test('it calls onDirectionChange handler when user clicks on sortable ascending cell', () => {
+  const onDirectionChange = jest.fn();
+  render(
+    <Table>
+      <Table.Header>
+        <Table.HeaderCell isSortable={true} onDirectionChange={onDirectionChange} sortDirection="ascending">
+          An header
+        </Table.HeaderCell>
+        <Table.HeaderCell>Another header</Table.HeaderCell>
+      </Table.Header>
+    </Table>
+  );
+
+  const header = screen.getByText('An header');
+  fireEvent.click(header);
+
+  expect(onDirectionChange).toBeCalledWith('descending');
+});
+
+test('it does not calls onDirectionChange handler when header is not sortable', () => {
+  const onDirectionChange = jest.fn();
+  render(
+    <Table>
+      <Table.Header>
+        <Table.HeaderCell isSortable={false} onDirectionChange={onDirectionChange} sortDirection="descending">
+          An header
+        </Table.HeaderCell>
+        <Table.HeaderCell>Another header</Table.HeaderCell>
+      </Table.Header>
+    </Table>
+  );
+
+  const header = screen.getByText('An header');
+  fireEvent.click(header);
+
+  expect(onDirectionChange).not.toBeCalled();
```

I'm wondering if we should throw an error when creating the table in this case